xla/service/gpu/buffer_comparator.cu.cc

```diff
@@ -108,6 +108,7 @@ __global__ void xla_fp8_e4m3fnuz_comparison(__hip_fp8_storage_t* buffer_a,
                                             float rel_error_threshold,
                                             uint64_t buffer_length,
                                             int* mismatch_count) {
+#if (defined(__gfx940__) || defined(__gfx941__) || defined(__gfx942__))
```

Comment:
Hi @ScXfjiang do we need to upstream this PR? if we are going to upstream it, could you use this [gfx9_mi200_or_later()](https://github.com/ROCm/xla/blob/rocm-jaxlib-v0.4.31-qa/xla/stream_executor/device_description.h#L184-L188) instead of this ugly macro everywhere? In addition, could you have this PR to 0.4.30-qa branch as well, please.

Reply:
Hi @i-chaochen, sorry for the late reply. Yes, we should because NANOO FP8 (those fp8 with `funz` suffix ) is only supported on MI300 devices. The PR (https://github.com/openxla/xla/pull/19896) has been submitted upstream and is waiting for the review.

Reply:
In addition, this change has already been merged into [rocm-jaxlib-v0.4.30-qa](https://github.com/ROCm/xla/tree/rocm-jaxlib-v0.4.30-qa). PR: https://github.com/ROCm/xla/pull/66